### ISSUE TITLE
Django `TIMEOUT` setting not working as expected

### ISSUE THREAD

**Issue by peterlopez** [jazzband/django-redis#713]
**Describe the bug**
Redis entries have a `-1` TTL value when using either Django's default `TIMEOUT` setting or custom `TIMEOUT` setting.

**To Reproduce**
1. Start with a completely empty Redis server.
2. Configure Django with or without `TIMEOUT` setting like so:

with:
```
CACHES = {
    'default': {
        'BACKEND': 'django_redis.cache.RedisCache',
        "LOCATION": "redis://'127.0.0.1:6379/1",
        'OPTIONS': {
            'CLIENT_CLASS': 'django_redis.client.DefaultClient',
        },
        'TIMEOUT': 1
    }
}
```

or without:
```
CACHES = {
    'default': {
        'BACKEND': 'django_redis.cache.RedisCache',
        "LOCATION": "redis://'127.0.0.1:6379/1",
        'OPTIONS': {
            'CLIENT_CLASS': 'django_redis.client.DefaultClient',
        }
    }
}
```

3. Load a page served by the Django app.
4. Inspect the Redis entry's TTL value using the `redis-cli TTL {key}` command.
5. See output of `-1` for TTL value of entry.

**Expected behavior**
I expected to see a Redis TTL value of either `300` when using the default Django `TIMEOUT` setting [as documented here](https://docs.djangoproject.com/en/3.2/ref/settings/#timeout) or `1`when I specify a `TIMEOUT` setting value of `1`.

**Stack trace**
Here's a screenshot of a debugger in case it helps.
<img width="1124" alt="Screenshot 2023-12-22 at 9 24 35 PM" src="https://github.com/jazzband/django-redis/assets/3279147/163323ca-e1a0-4d89-9464-f4a7ef07e3fd">
<img width="1137" alt="Screenshot 2023-12-22 at 10 08 57 PM" src="https://github.com/jazzband/django-redis/assets/3279147/d65ba315-e615-44ca-bade-0a532fda79a8">

**Environment (please complete the following information):**
 - Python version: 3.7
 - Django Redis Version: 4.12.1
 - Django Version: 3.2
 - Redis Version: 7.2.2
 - redis-py Version: 5.0.1

**Additional context**
I'm looking after a client's outdated Django site that uses Wagtail CMS. I noticed their Redis cache is slowly filling up since all keys have TTL values of `-1`. Happy to upgrade `django-redis` if needed or use an alternative package if this one does not support the `TIMEOUT` setting. Perhaps Wagtail CMS has the issue since nowhere in the codebase could I find calls directly to this package's `cache.set()` interface.

**Comment by WisdomPill** [comment]
Hello, timeout is supported and used heavily by myself as well, there are tests for it as well, so my question is, could you provide a more detailed description? How is `django-redis` used by this CMS library you were talking about? Could you provide an example code to replicate the behavior you're seeing?

**Comment by peterlopez** [comment]
@WisdomPill thanks for the quick response! After doing a bit more digging, I discovered the app is using the `django-cachalot` package while the [`CACHALOT_TIMEOUT` setting](https://django-cachalot.readthedocs.io/en/latest/quickstart.html#cachalot-timeout) is not configured. Thus the package's default value of `None` was being used for the TTL. After adding the `CACHALOT_TIMEOUT` setting, I can see the TTL value is reflected by `redis-cli` as I expect.

There is no bug in `django-redis`. Closing this issue.
